src/rsa.rs

```diff
@@ -27,6 +28,30 @@ pub struct RSAPKCS1Padding {
     digestinfo_prefix: &'static [u8],
 }
 
+impl RSAPKCS1Padding {
+    fn pad(&self, msg: &[u8], out: &mut [u8]) -> Result<(), ()> {
+        let digest_len = self.digestinfo_prefix.len() + self.digest_alg.output_len;
+        if out.len() < digest_len + 11 {
```

Comment:
Actually, this is dead code, at least as far as our test suite goes, right?

I think instead, we should have `rsa_verify_end()` reject keys where `n < 2048`, and then change this check to a `debug_assert!()` because we reject small keys.

Consider: a SHA-512 hash is 64 bytes. The digest_info prefix is 19 bytes. 64 + 19 + 11 = 94. 94 * 8 bits per byte =  752 bits.

Otherwise, we need to make a test vector with a <752 bit key for testing this.